### PR TITLE
[CHK-1681] handling new data for transaction details in close payment request

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Update close payment request body according to the new specs about transaction details fields

#### List of Changes

Update close payment request body building adding all required data for biz event

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -413,8 +413,8 @@
                 <configuration>
                     <skipCheckoutIfExists>false</skipCheckoutIfExists>
                     <connectionUrl>scm:git:https://github.com/pagopa/pagopa-ecommerce-commons.git</connectionUrl>
-                    <scmVersionType>tag</scmVersionType>
-                    <scmVersion>${pagopa-ecommerce-commons.version}</scmVersion>
+                    <scmVersionType>branch</scmVersionType>
+                    <scmVersion>CHK-1674-add-pspOnUs-flag</scmVersion>
```
Comment (giovanniberti) [inline pom.xml:417]: Add dependency on the corresponding PR

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java [hunk 2]
```diff
@@ -193,56 +172,22 @@ public Mono<Tuple2<Optional<TransactionRefundRequestedEvent>, Either<Transaction
...
+                                    authRequestData,
+                                    transactionAuthorizationRequestData,
+                                    transactionAuthorizationCompletedData,
+                                    command,
+                                    fee,
+                                    amount,
+                                    totalAmount,
+                                    outcomeEnum
+                            )
+                    );
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java:190]: This doesn't modify the original object but creates a new one. Is it necessary to split the method call?

**Review by giovanniberti — CHANGES_REQUESTED** [review]

**Comment by giovanniberti** [comment]
Also remember to update API tests if necessary :)

**Comment by ciuffagianluca** [comment]
Depends on https://github.com/pagopa/pagopa-ecommerce-commons/pull/95

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java [hunk 3]
```diff
@@ -411,6 +361,125 @@ or at (start + paymentTokenTimeout - offset) where `start` is the creation time
...
+                .transactionId(
+                        Optional.ofNullable(command.getData()).get().transaction()
+                                .getTransactionId().value()
+                )
+                .creationDate(
+                        Optional.ofNullable(command.getData()).get().transaction()
+                                .getCreationDate().toOffsetDateTime()
+                )
+                .paymentGateway(
+                        Optional.ofNullable(transactionAuthorizationRequestData).get().getPaymentGateway().name()
```
Comment (Fernando-Granato) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java:420]: why do you check for the correct value of transactionAuthorizationRequestData ? 
```suggestion
                        transactionAuthorizationRequestData.getPaymentGateway().name()
```

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java [hunk 4]
```diff
@@ -411,6 +361,125 @@ or at (start + paymentTokenTimeout - offset) where `start` is the creation time
...
+                )
+                .rrn(Optional.ofNullable(authRequestData).get().rrn())
+                .authorizationCode(Optional.ofNullable(authRequestData).get().authorizationCode())
+                .timestampOperation(
+                        Optional.ofNullable(transactionAuthorizationCompletedData).get().getTimestampOperation()
+                )
+                .errorCode(
+                        ClosePaymentRequestV2Dto.OutcomeEnum.KO.equals(outcomeEnum)
+                                ? Optional.ofNullable(transactionAuthorizationCompletedData).get().getErrorCode()
```
Comment (Fernando-Granato) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java:429]: Same for this!

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java [hunk 5]
```diff
@@ -411,6 +361,125 @@ or at (start + paymentTokenTimeout - offset) where `start` is the creation time
...
+    ) {
+        return new TransactionDto()
+                .transactionStatus(
+                        ClosePaymentRequestV2Dto.OutcomeEnum.OK.equals(outcomeEnum) ? "Autorizzato" : "Rifiutato"
+                )
+                .fee(fee)
+                .amount(amount)
+                .grandTotal(totalAmount)
+                .transactionId(
+                        Optional.ofNullable(command.getData()).get().transaction()
```
Comment (Fernando-Granato) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java]: Remove this option check?

File: src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java [hunk 6]
```diff
@@ -411,6 +361,125 @@ or at (start + paymentTokenTimeout - offset) where `start` is the creation time
...
+                                .getTransactionId().value()
+                )
+                .creationDate(
+                        Optional.ofNullable(command.getData()).get().transaction()
```
Comment (Fernando-Granato) [inline src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java]: Remove this option check?

**Review by Fernando-Granato — APPROVED** [review]

**Pull request merged**